merkle/smt/tile.go

```diff
@@ -28,3 +28,34 @@ type Tile struct {
 	ID     tree.NodeID2
 	Leaves []Node
 }
+
+// scan visits all non-empty nodes of the tile except the root.
+func (t Tile) scan(lo *tree.Layout, h mapHasher, visit func(tree.NodeID2, []byte)) error {
+	top := t.ID.BitLen()
+	height := uint(lo.TileHeight(int(top)))
+	// TODO(pavelkalinnikov): Remove HStar3 side effects, to avoid copying here.
```

Comment:
Is there a link to context here? It's OK if this makes sense to you and you plan to resolve it soon, but if you got abducted by aliens this weekend I would need to put in some work to find what side-effect this was pointing at and how/why it might be changed.